pkg/trace/api/container_linux.go

```diff
@@ -46,17 +46,27 @@ func connContext(ctx context.Context, c net.Conn) context.Context {
 	if !ok {
 		return ctx
 	}
-	file, err := s.File()
+	raw, err := s.SyscallConn()
 	if err != nil {
-		log.Debugf("Failed to obtain unix socket file: %v", err)
+		log.Debugf("Failed to read credentials from unix socket: %v", err)
 		return ctx
 	}
-	fd := int(file.Fd())
-	ucred, err := syscall.GetsockoptUcred(fd, syscall.SOL_SOCKET, syscall.SO_PEERCRED)
+	var (
+		ucred *syscall.Ucred
+		cerr  error
+	)
+	err = raw.Control(func(fd uintptr) {
+		ucred, cerr = syscall.GetsockoptUcred(int(fd), syscall.SOL_SOCKET, syscall.SO_PEERCRED)
+	})
 	if err != nil {
 		log.Debugf("Failed to read credentials from unix socket: %v", err)
 		return ctx
 	}
+	if cerr != nil {
+		log.Debugf("Failed to read credentials from unix socket: %v", cerr)
```

Comment:
should this error message be the same as the one above or can a different error happen here?

Reply:
Yeah, I can differentiate them.